include/seqan3/io/structure/structure_file_in.hpp

```diff
@@ -0,0 +1,1217 @@
+            if constexpr (selected_field_ids::contains(field::REACT_ERR))
+                react_err_column_buffer.push_back(std::move(seqan3::get<field::REACT_ERR>(rec)));
+            if constexpr (selected_field_ids::contains(field::COMMENT))
+                comment_column_buffer.push_back(std::move(seqan3::get<field::COMMENT>(rec)));
+            if constexpr (selected_field_ids::contains(field::OFFSET))
+                offset_column_buffer.push_back(std::move(seqan3::get<field::OFFSET>(rec)));
+        }
+    }
+
+    //!\brief Befriend iterator so it can access the buffers.
+    friend iterator;
+};
+
+/*!\name Type deduction guides
+ * \relates seqan3::structure_file_in
+ * \{
+ */
+template <istream_concept<char> stream_type,
+          structure_file_in_format_concept file_format,
+          detail::fields_concept selected_field_ids>
+structure_file_in(stream_type && _stream, file_format const &, selected_field_ids const &)
+    -> structure_file_in<typename structure_file_in<>::traits_type,       // actually use the default
+                         selected_field_ids,
+                         type_list<file_format>,
+                         std::remove_reference_t<stream_type>>;
+//!\}
+
+} // namespace seqan3
+
+// ------------------------------------------------------------------
+// std-overloads for the tuple-like interface
+// ------------------------------------------------------------------
+
+namespace std
+{
+//!\brief std::tuple_size overload for column-like access. [metafunction specialisation for seqan3::structure_file_in]
+template<seqan3::structure_file_in_traits_concept traits_type,
+         seqan3::detail::fields_concept selected_field_ids,
+         seqan3::detail::type_list_of_structure_file_in_formats_concept valid_formats,
+         seqan3::istream_concept<char> stream_type>
+struct tuple_size<seqan3::structure_file_in<traits_type, selected_field_ids, valid_formats, stream_type>>
+{
+    //!\brief The value equals the number of selected fields in the file.
+    static constexpr size_t value = selected_field_ids::as_array.size();
+};
+
+//!\brief std::tuple_element overload for column-like access. [metafunction specialisation for seqan3::structure_file_in]
+template<size_t elem_no,
+         seqan3::structure_file_in_traits_concept traits_type,
+         seqan3::detail::fields_concept selected_field_ids,
+         seqan3::detail::type_list_of_structure_file_in_formats_concept valid_formats,
+         seqan3::istream_concept<char> stream_type>
+struct tuple_element<elem_no, seqan3::structure_file_in<traits_type, selected_field_ids, valid_formats, stream_type>>
+    : tuple_element<elem_no, typename seqan3::structure_file_in<traits_type,
+                                                                selected_field_ids,
+                                                                valid_formats,
+                                                                stream_type>::file_as_tuple_type>
```

Comment:
I remember we fixed something here, can you make a PR for sequence_io or at least open an issue so we don't forget? Thanks!

Reply:
Yes, I found some more stuff and I already have a separate branch for that.

Reply:
see PR #313